src/MediaCollections/FileAdder.php

```diff
@@ -431,6 +432,8 @@ protected function processMediaItem(HasMedia $model, Media $media, self $fileAdd
 
         $this->checkGenerateResponsiveImages($media);
 
+        DB::beginTransaction();
```

Comment:
I wonder if this isn't breaking. What would happen if the user already started a transaction?

Reply:
@freekmurze I had the same thought, perhaps there's a better way to approach this functionality without using transactions.  

@Silver343, what are your thoughts on this?

Reply:
@freekmurze @patinthehat I believe Laravel handles nested transactions by using a  [transaction counter](https://github.com/laravel/framework/blob/9.x/src/Illuminate/Database/Concerns/ManagesTransactions.php), but I may be misunderstanding it.

If the functionality could be created without using transactions that may be preferable.

Reply:
I refactored and removed this; see comment below.